articles/200_migration_guide_from_ros1.md

```diff
@@ -145,6 +199,34 @@ Shared pointer types are provided as typedefs within the message structs: `my_in
 
 For more details please see the article about the [generated C++ interfaces](http://design.ros2.org/articles/generated_interfaces_cpp.html).
 
+The migration requires includes to change by:
+
+* insert the subfolder `msg` between the package name and message datatype
+* Change the included filename from CamelCase to underscore separation
+* Change from `*.h` to `*.hpp`
+
+The migration requires code to insert the `msg` namespace into all instances.
+
+
+#### Usages of ros::Time
+*TODO(tfoote) There is no direct replacement for ros::Time yet we expect to have one in the future.*
+
+Under the hood we expect to leverage the cross platform `std::chrono` library.
+
+Currently for usages of `ros::Time`:
+
+* replace all instances of `ros::Time` with `builtin_interfaces::msg::Time`
+* Convert all instances of `nsec` to `nanosec`
+* Convert all single argument double constructors to bare constructor + assignment
+
+Field values do not get initialized to zero when constructed.
+You must make sure to set all values instead of relying on them to be zero.
```

Comment:
@dirk-thomas Should we change this? The generated CPP messages have a constructor which set the default values, but not if there is no explicit default value. Should fields without an explicitly specified default value get the default value for the field type, e.g. 0 for ints, 0.0 for floats, '' for string, False for bool, etc...